### DISCUSSION TITLE
Node.js version in user vs maintainer context

### DISCUSSION THREAD

**Discussion by yoriiis** [yoriiis/svg-chunk-webpack-plugin#13] [General]
@alexander-akait I need clarification about Node.js version in user vs maintainer context.

The `package-lock.json` should be generated with which version of Node.js?

For example, the project supports `v12+` (the `engines.node` indicates it) but I want to update the dependencies and play the tests in CI with newer versions (all upcoming LTS releases).

I tried to use `v18` for the development on the project, and generated a `package-lock.json` with this version (which seems to me to be the right way to proceed) but I have an error related to the package `schema-utils` with the following scenario:

- The `package-lock.json` was generated with `v18`
- CI run `npm ci` with `v14`
- The following error appears `npm ERR! Cannot read property 'schema-utils' of undefined` (see https://github.com/yoriiis/chunks-webpack-plugin/actions/runs/4747514854/jobs/8432521463#step:4:5)

Other scenario:

- The `package-lock.json` was generated with `v18`
- CI run `npm ci` with `v12`
- The `css-minimizer-webpack-plugin` trigger an error (see https://github.com/yoriiis/svg-chunk-webpack-plugin/pull/12/checks#step:9:16)

I must be missing something sorry. What would you recommend?

**Comment by alexander-akait** [comment]
> The following error appears npm ERR! Cannot read property 'schema-utils' of undefined

It is npm bug, lockfiles are not compatibility between versions (new npm versions support new and old lockfiles, old npm version doesn't support new lockfiles), if I need support to node v12 I generate lockfile using node v12 (nvm for switching between versions)

> The css-minimizer-webpack-plugin trigger an error (see https://github.com/yoriiis/svg-chunk-webpack-plugin/pull/12/checks#step:9:16)

This can't be fixed, it is a new syntax, you can disable this plugin for tests if you really want to support node v12, but I recommend don't do it, drop v12 and v14, no need to support old engines
